### ISSUE TITLE
RS485 support for IM uplink

### ISSUE THREAD

**Issue by kenthartig** [kenthartig/Pro3-Linux#18]
The enable pins for the RS485 transceiver are supposedly supported natively in the Linux kernel. I have turned on this support but I think the DTSI file needs to be modified to support the two I/O pins.

**Comment by paholden** [comment]
This is normally the case.  The Linux distribution includes the support, but menuconfig and device tree need to be set up to enable them.

**Comment by kenthartig** [comment]
Yes. menuconfig has been enabled but I believe the device tree needs to establish the RTS pin in PINMUX.

**Comment by kenthartig** [comment]
&uart1  {
	status = "okay";
	label = "RS-485 link";
	compatible = "ti,omap4-uart";
	reg = <0x4806a000 0x100>;
	interrupts-extended = <&crossbar_mpu GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>;
	ti,hwmods = "uart1";
	clock-frequency = <48000000>;
	linux,rs485-enabled-at-boot-time;		/* KH Enable RS-485 mode */
	rs485-rts-active-high;				/* KH select high to transmit*/
    	rs485-rx-during-tx;				/* default full duplex */
	rs485-rts-delay = <0 200>;			/* milliseconds */
};

**Comment by kenthartig** [comment]
I have added GPIO support as well. The hardware can be strapped to either use the discrete GPIOs or the rts pin.

**Comment by tkuschel** [comment]
@kenthartig 

the line inside the device tree for the &uart1 should be:
```
ti,hwmods = "uart2";   /* as I have seen and tested, the hwmods parameter is always 1 higher */
```

**Comment by paholden** [comment]
Have you actually seen a problem with this?

**Comment by kenthartig** [comment]
I don't even know how this guy could see our repo.

I wonder if he may be a troll?

-Kent
________________________________

**Comment by tkuschel** [comment]
> I wonder if he may be a troll?
Unfortunatly the enable pins for the RS485 transceiver which should be supported natively in the Linux kernel do _not_ work with latest LTS Kernels 4.19 / 5.4.
I just wanted to briefly point out that since installing the OMAP driver into the generic 8280 linux driver ("Support for OMAP internal UART (8250 based driver), the functionality of the RS485-DE (beagle bone black) no longer works (tested with 4.19.165 Kernel and also 5.4.96). My RS-485 only works with the "old" OMAP serial driver (OMAP serial port support) which is not selected per default (beagle bone black bb-kernel); and also the listed device tree file above has an entry "ti,hwmods" which internally has not the same numbering scheme of UARTs as TI described. A working implementation: 
https://github.com/RobertCNelson/bb-kernel/issues/38#issuecomment-770122621
(I found your open issue because my RS485 didn't work before, I searched on github for "RS-485" and finally saw your proposal of the device tree).

**Comment by paholden** [comment]
Is this the same guy that sent email a while back?  You need to privatize the repo.

**Comment by kenthartig** [comment]
Yes

Sent from my iPhone 8


On Feb 13, 2021, at 10:24 AM, Patricia Holden <notifications@github.com> wrote:

﻿

Is this the same guy that sent email a while back? You need to privatize the repo.

**Comment by paholden** [comment]
I don’t get why he continues to email you.

**Comment by kenthartig** [comment]
Thank you for your input Thomas. We'll try that out.

**Comment by paholden** [comment]
You are still getting emails from this guy?

**Comment by kenthartig** [comment]
No more

Sent from my iPhone 8


On Feb 26, 2021, at 12:55 PM, Patricia Holden <notifications@github.com> wrote:

﻿

You are still getting emails from this guy?
